### PR TITLE
Feature/eodhp 641 stopping workflow jobs

### PR DESCRIPTION
**Pull request by tjellicoe-tpzuk**
## Merge in Latest Upstream Commits
- Update how log URLs are generated, now including `deployed_workspace`
- Ensure logs are generated even when workflow fails

### CODE REVIEW

File: {{cookiecutter.service_name}}/service.py [hunk 1]
```diff
@@ -510,7 +512,13 @@ def {{cookiecutter.workflow_id |replace("-", "_")  }}(conf, inputs, outputs): #
 
     except Exception as e:
         logger.error("ERROR in processing execution template...")
+        logger.error("Try fetching logs if any...")
+        try:
+            tool_logs = runner.execution.get_tool_logs()
+            execution_handler.handle_outputs(None, None, None, tool_logs)
+        except Exception as e:
+            logger.error("Fethcing logs failed!"+str(e))
```
Comment (apalmer-tpzuk) [inline {{cookiecutter.service_name}}/service.py]: Typo - Fetching

File: {{cookiecutter.service_name}}/service.py [hunk 2]
```diff
@@ -510,7 +512,13 @@ def {{cookiecutter.workflow_id |replace("-", "_")  }}(conf, inputs, outputs): #
...
         stack = traceback.format_exc()
         logger.error(stack)
         conf["lenv"]["message"] = zoo._(f"Exception during execution...\n{stack}\n")
-        return zoo.SERVICE_FAILED
+        return zoo.SERVICE_FAILED
```
Comment (apalmer-tpzuk) [inline {{cookiecutter.service_name}}/service.py]: Missing empty line at end of file

File: {{cookiecutter.service_name}}/service.py [hunk 3]
```diff
@@ -510,7 +512,13 @@ def {{cookiecutter.workflow_id |replace("-", "_")  }}(conf, inputs, outputs): #
 
     except Exception as e:
         logger.error("ERROR in processing execution template...")
+        logger.error("Try fetching logs if any...")
```
Comment (apalmer-tpzuk) [inline {{cookiecutter.service_name}}/service.py]: I think this should become logger.info as it is not indicative of an error itself

File: {{cookiecutter.service_name}}/service.py (hunk 1 above)
Comment (tjellicoe-tpzuk) [reply {{cookiecutter.service_name}}/service.py]: fixed

File: {{cookiecutter.service_name}}/service.py (hunk 2 above)
Comment (tjellicoe-tpzuk) [reply {{cookiecutter.service_name}}/service.py]: Added

File: {{cookiecutter.service_name}}/service.py (hunk 3 above)
Comment (tjellicoe-tpzuk) [reply {{cookiecutter.service_name}}/service.py]: updated

**Review by apalmer-tpzuk — APPROVED** [review]

**Pull request merged**